src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/Serialization/TagHelperDescriptorJsonConverter.cs

```diff
@@ -906,5 +914,141 @@ private static void ReadToEndOfCurrentObject(JsonReader reader)
 
             throw new JsonSerializationException($"Could not read till end of object, end of stream. Got '{reader.TokenType}'.");
         }
+
+        private static string Intern(string str)
+        {
+            if (str is null)
+            {
+                return null;
+            }
+
+            var interned = string.IsInterned(str);
+            if (interned != null)
+            {
+                return interned;
+            }
+
+            return weakReferenceTable.Intern(str);
+        }
+
+        /// <summary>
+        /// The purpose of this class is to avoid permanently storing duplicate strings which were read in from JSON.
+        /// </summary>
+        private class WeakReferenceTable
```

Comment:
We may later want to pull this out into its own class with tests and all that if we want to do something similar elsewhere, and a quick glance suggests we may want to.